src/js/actions/userActions.js

```diff
@@ -571,21 +571,33 @@ export const saveGlobalSettings =
       });
   };
 
-export const saveUserSettings = settings => (dispatch, getState) => {
-  if (!getState().users.currentUser) {
-    return Promise.resolve();
-  }
-  return Promise.resolve(dispatch(getGlobalSettings())).then(() => {
-    const userSettings = getUserSettings(getState());
-    const updatedSettings = {
-      [getState().users.currentUser]: {
-        ...userSettings,
-        ...settings
-      }
-    };
-    return Promise.resolve(dispatch(saveGlobalSettings(updatedSettings)));
+export const getUserSettings = () => dispatch =>
+  GeneralApi.get(`${useradmApiUrl}/settings/me`).then(({ data: settings, headers: { etag } }) => {
+    window.sessionStorage.setItem(UserConstants.settingsKeys.initialized, true);
+    return Promise.all([dispatch({ type: UserConstants.SET_USER_SETTINGS, settings }), etag]);
   });
-};
+
+export const saveUserSettings =
+  (settings = { onboarding: {} }) =>
+  (dispatch, getState) => {
+    if (!getState().users.currentUser) {
+      return Promise.resolve();
+    }
+    return Promise.resolve(dispatch(getUserSettings())).then(result => {
+      const userSettings = getUserSettingsSelector(getState());
+      const updatedSettings = {
+        ...userSettings,
+        ...settings,
+        onboarding: {
+          ...userSettings.onboarding,
+          ...settings.onboarding
+        }
+      };
+      return GeneralApi.post(`${useradmApiUrl}/settings/me`, updatedSettings, { headers: { etag: result[result.length - 1] } }).then(() =>
```

Comment:
This is wrong: you should use "If-Match: ETAG", where ETAG is the previous value. However, you can insert the header only if you have a value for ETAG. In other words, you cannot send "undefined".